pkg/config/remote/service/service.go

```diff
@@ -1117,29 +1114,14 @@ func (c *HTTPClient) getUpdate(
 	if tufVersions.DirectorTargets == currentTargetsVersion {
 		return nil, nil
 	}
-	roots, err := c.getNewDirectorRoots(c.uptane, currentRootVersion, tufVersions.DirectorRoot)
-	if err != nil {
-		return nil, err
-	}
-	targetsRaw, err := c.uptane.TargetsMeta()
-	if err != nil {
-		return nil, err
-	}
-	targetFiles, err := c.getTargetFiles(c.uptane, rdata.StringListToProduct(products), cachedTargetFiles)
-	if err != nil {
-		return nil, err
-	}
-
-	canonicalTargets, err := enforceCanonicalJSON(targetsRaw)
-	if err != nil {
-		return nil, err
-	}
 
+	// Filter out files that either:
```

Comment:
teeny tiny nit: I would prefer a named helper function here to explain the behavior of the function rather than comments since comments are easier to forget about/remove if/when this behavior is changed or removed

Reply:
Agreed that both areas where config updates are fetched could use refactoring, but I'm not going to introduce that into the scope of this PR if that's ok. I could have even left this comment out, it's not describing any new behavior as it is doing exactly what it did before. Just filtering BEFORE we ask for raw target files rather than getting all of them, and filtering.